src/rustup-cli/rustup_mode.rs

```diff
@@ -687,7 +688,15 @@ fn toolchain_link(cfg: &Cfg, m: &ArgMatches) -> Result<()> {
     let ref toolchain = m.value_of("toolchain").expect("");
     let ref path = m.value_of("path").expect("");
     let toolchain = try!(cfg.get_toolchain(toolchain, true));
-
+    let mut pathbuf = PathBuf::from(path);
+
+    pathbuf.push("lib");
+    try!(utils::assert_is_directory(&pathbuf));
+    pathbuf.pop();
+    pathbuf.push("bin");
+    try!(utils::assert_is_directory(&pathbuf));
+    pathbuf.push("rustc");
+    try!(utils::assert_is_file(&pathbuf));
```

Comment:
@brson Done. I've added it to the top of the function as `install_from_dir` may be used in the future with `link=false` but it would still require checking the directory structure correctly in that case.